sage/embedder.py

```diff
@@ -335,6 +344,67 @@ def download_embeddings(self) -> Generator[Vector, None, None]:
         return []
 
 
+class GeminiBatchEmbedder(BatchEmbedder):
+    """Batch embedder that calls Gemini."""
+
+    def __init__(self, data_manager: DataManager, chunker: Chunker, embedding_model: str):
+        self.data_manager = data_manager
+        self.chunker = chunker
+        self.embedding_data = []
+        self.embedding_model = embedding_model
+        genai.configure(api_key=os.environ["GEMINI_API_KEY"])
+
+    def _make_batch_request(self, chunks: List[Chunk]) -> Dict:
+        return genai.embed_content(
+            model=self.embedding_model,
+            content=[chunk.content for chunk in chunks],
+            task_type="retrieval_document")
+
+    def embed_dataset(self, chunks_per_batch: int, max_embedding_jobs: int = None):
+        """Issues batch embedding jobs for the entire dataset."""
+        batch = []
+        chunk_count = 0
+
+        for content, metadata in self.data_manager.walk():
+            chunks = self.chunker.chunk(content, metadata)
+            chunk_count += len(chunks)
+            batch.extend(chunks)
+
+            token_count = chunk_count * self.chunker.max_tokens
+            if token_count % 900_000 == 0:
+                logging.info("Pausing for 60 seconds to avoid rate limiting...")
+                time.sleep(60)
```

Comment:
Did you just copy this from Voyage? Not all providers have the same rate limits (e.g. OpenAI doesn't). It's worth checking because this is a pretty big bottleneck on speed and we don't want to throttle things unnecessarily.